### ISSUE TITLE
Unclear grounds for Yoda intervention

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#15188]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Style/YodaCondition

### Steps to reproduce the problem

```ruby
def foo?(*)
  [*] == [*].compact # Discovered via
end

[a] == [a].b # Or simply
```

This form is not mentioned in https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/YodaCondition

### Expected behavior

Intuitively this doesn't seem comparable(!) to `42 == life`. Do elaborate if there is a rationale. And regardless:

```
diff --git a/lib/rubocop/cop/style/yoda_condition.rb b/lib/rubocop/cop/style/yoda_condition.rb
index 86020da89..02c52b57d 100644
--- a/lib/rubocop/cop/style/yoda_condition.rb
+++ b/lib/rubocop/cop/style/yoda_condition.rb
@@ -81,3 +81,3 @@ module RuboCop
 
-        MSG = 'Reverse the order of the operands `%<source>s`.'
+        MSG = 'Reverse the order of the operands `%<source>s` you must.'
         REVERSE_COMPARISON = { '<' => '>', '<=' => '>=', '>' => '<', '>=' => '<=' }.freeze
```

### Actual behavior

```
Reverse the order of the operands `[*] == [*].compact`. [[Correctable] Style/YodaCondition]
```

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
System Version: macOS 26.5 (25F71)
Model Identifier: MacBookAir10,1
ruby 4.0.5 (2026-05-20 revision 64336ffd0e) +PRISM [arm64-darwin25]
rubocop     2026-05-27 4a12596a3
```
